File: wow-minmax/src/main/java/wow/minmax/model/CharacterStats.java
package wow.minmax.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import wow.commons.model.spells.SpellSchool;

import java.util.Map;

/**
 * User: POlszewski
 * Date: 2023-01-01
 */
@Data
@AllArgsConstructor
public class CharacterStats {
	private double sp;
	private Map<SpellSchool, Double> spellDamageBySchool;
	private double hitRating;
	private double hitPct;
	private double critRating;
	private double critPct;
	private double hasteRating;
	private double hastePct;
	private double stamina;
	private double intellect;
	private double spirit;
}

File: wow-minmax-ui/src/app/modules/statistics/model/CharacterStats.ts
import { SpellSchool } from "../../shared/model/spell/SpellSchool";

export interface CharacterStats {
	type: string;
	sp: number;
	spellDamageBySchool: Partial<Record<SpellSchool, number>>;
	hitRating: number;
	hitPct: number;
	critRating: number;
	critPct: number;
	hasteRating: number;
	hastePct: number;
	stamina: number;
	intellect: number;
	spirit: number;
	maxHealth: number;
	maxMana: number;
}
